ies-ruby/providers/rubies.rb

```diff
@@ -0,0 +1,126 @@
+def rbenv_user
+  @rbenv_user ||= new_resource.rbenv_user
+  @rbenv_user ||= 'root'
+end
+
+def home
+  @home ||= node['etc']['passwd'][rbenv_user]['dir']
+end
+
+def rbenv_home
+  @rbenv_home ||= "#{home}/.rbenv"
+end
+
+def rbenv_revision
+  @rbenv_revision ||= new_resource.rbenv_revision
+  @rbenv_revision ||= 'master'
+end
+
+def ruby_build_revision
+  @ruby_build_revision ||= new_resource.ruby_build_revision
+  @ruby_build_revision ||= 'master'
+end
+
+def opsworks_mandatory_ruby
+  @opsworks_mandatory_ruby ||= new_resource.opsworks_mandatory_ruby
+end
+
+action :install do
+  desired_ruby = new_resource.desired_ruby
+
+  if Dir.exist?("#{rbenv_home}/versions/#{desired_ruby}")
+    Chef::Log.info("#{desired_ruby} is already installed, skipping!")
+  else
```

Comment:
Please remove the else block, and use `next` in the if statement, makes it easier to read.